instrumentation/micrometer/micrometer-1.5/library/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/OpenTelemetryMeterRegistry.java

```diff
@@ -53,13 +53,18 @@ public static OpenTelemetryMeterRegistryBuilder builder(OpenTelemetry openTeleme
   private final io.opentelemetry.api.metrics.Meter otelMeter;
 
   OpenTelemetryMeterRegistry(
-      Clock clock, TimeUnit baseTimeUnit, io.opentelemetry.api.metrics.Meter otelMeter) {
+      Clock clock,
+      TimeUnit baseTimeUnit,
+      boolean prometheusMode,
+      io.opentelemetry.api.metrics.Meter otelMeter) {
     super(clock);
     this.baseTimeUnit = baseTimeUnit;
     this.otelMeter = otelMeter;
 
+    NamingConvention namingConvention =
+        prometheusMode ? PrometheusModeNamingConvention.INSTANCE : NamingConvention.identity;
```

Comment:
By the way random note, I noticed our shim defaults to `identity` but micrometer seems to default to `snake_case` IIUC. Any reason ours is different?

https://github.com/micrometer-metrics/micrometer/blob/main/micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java#L142

Reply:
I guess because we expect the prometheus exporter to do the conversion

Reply:
Yep, that's one of the reasons. The other is that names using dots as the "word separator" seemsto match the spec names the closest (the spec doesn't explicitly mention concatenating with dots as the official strategy, but all instrument names follow this)